pillar/edx/sandbox.sls

```diff
@@ -28,6 +32,24 @@ edx:
     EDXAPP_MYSQL_CSMH_USER: __vault__:cache:mysql-{{ environment }}/creds/edxapp-csmh-{{ purpose }}>data>username
     EDXAPP_MEMCACHE:
       - 'localhost:11211'
+    ################################################################################
+    #################### Forum Settings ############################################
+    ################################################################################
+    FORUM_API_KEY: __vault__:gen_if_missing:secret-residential/global/forum-api-key>data>value
```

Comment:
This technically isn't residential so we should probably use a different secret mount (`secret-mitxpro`). It might also be worth thinking about moving to a per-env key but not necessary at the moment.

As an aside, we should check if we're using the residential mount for other secrets in the xpro env.

Reply:
We also use it in the `cloud_deployment` pillar which applies to the xpro env

Reply:
Yeah, I thought that might be the case... We can change that in a separate PR